include/cantera/kinetics/ThirdBodyCalc.h

```diff
@@ -104,7 +106,35 @@ class ThirdBodyCalc3
             AssertTrace(eff.first != npos);
             m_species.back().push_back(eff.first);
             m_eff.back().push_back(eff.second - default_efficiency);
+            m_efficiencyList.emplace_back(
+                rxnNumber, eff.first, eff.second - default_efficiency);
+        }
+    }
+
+    //! Resize the sparse coefficient matrix
+    void resizeCoeffs(size_t nSpc, size_t nRxn) {
+        // Sparse Efficiency coefficient matrix
+        Eigen::SparseMatrix<double> efficiencies;
+        efficiencies.setZero();
+        efficiencies.resize(nRxn, nSpc);
+        efficiencies.reserve(m_efficiencyList.size());
+        efficiencies.setFromTriplets(
+            m_efficiencyList.begin(), m_efficiencyList.end());
+
+        // Jacobian matrix multipliers
+        std::vector<Eigen::Triplet<double>> triplets;
+        triplets.reserve(nRxn * nSpc);
```

Comment:
Yes. But I'm not sure how to estimate as in theory the matrix could be dense. There likely is a way to optimize - although per Perini et al (2012) the sparsity appears to be hard to predict; Table 1 gives examples for sparsity being anywhere between 46 and 98%.

Reply:
Building a `std::vector` element-by-element is always just an O(n) operation, even if it invokes some internal resize operations. If there's an easy way to know how much space is needed ahead of time, it can save a small amount of time, but I think making a much larger than necessary allocation could actually backfire, and it would be better to just take the simpler route of not reserving anything.